nightfall-client/src/event-handlers/block-proposed.mjs

```diff
@@ -57,14 +64,37 @@ async function blockProposedEventHandler(data) {
   await Promise.all(dbUpdates);
   const updatedTimber = Timber.statelessUpdate(latestTree, blockCommitments);
   await saveTree(transactionHashL1, block.blockNumberL2, updatedTimber);
+
+  // We update all sibling paths to account for the newly inserted leaves.
+  const currentPaths = await getAllSiblingPaths();
```

Comment:
Would this attempt to pull all sibling paths from the DB each time a block is proposed and process them all?  Is there an issue when we have millions of commits? I'm worried that the processing of a block event will become too slow.

Reply:
Fair enough

Reply:
If it's restricted to just the commits that the client is concerned with then it's not really an issue.

Reply:
Yep it will. This will definitely slow down block event processing but unfortunately sibling path updates need to happen every block.

There are a couple of ways to mitigate this (dumping them here for reference later)

- Optimisations we can (need to) do
    1. Improve the performance of the sibling path update (i.e the batching and "dead-heading" techniques I described in the show-n-tell).
    2. Only update the sibling paths of commitments "in-play" (i.e. inside the challenge window)
    3. More efficient storage and retrieval patterns for updated paths.

- We can decide on `MAX_UPDATES` to be performed on a sibling path based on some analysis of the anonymity set. In my mind, there should be a "sweet spot" where "I have updated my sibling path for x number of blocks and there are y total blocks in the blockchain" such that the improvement in anonymity of updating to x+1 is minimal.

- For the client/browser slow block processing isn't terminal, you just might take longer to notice when you have received or successfully spent a commitment. 

I agree that if a user owns a million commitments today then updating the siblingPath of all these commitments will be probably be slow.